src/back/hlsl/writer.rs

```diff
@@ -72,42 +92,267 @@ impl<W: Write> Writer<W> {
         self.names.clear();
         self.namer.reset(module, RESERVED, &[], &mut self.names);
         self.named_expressions.clear();
+        self.ep_input = None;
     }
 
     pub fn write(&mut self, module: &Module, info: &ModuleInfo) -> BackendResult {
-        if self.shader_model < ShaderModel(50) {
-            return Err(Error::UnsupportedShaderModel(self.shader_model));
+        if self.options.shader_model < ShaderModel::default() {
+            return Err(Error::UnsupportedShaderModel(self.options.shader_model));
         }
 
+        // Try to find the entry point and corresponding index
+        let ep_idx = module
+            .entry_points
+            .iter()
+            .position(|ep| {
+                self.options.shader_stage == ep.stage && self.options.entry_point == ep.name
+            })
+            .ok_or(Error::EntryPointNotFound)?;
+
         self.reset(module);
 
-        // Write all entry points
-        for (index, ep) in module.entry_points.iter().enumerate() {
-            let func_ctx = FunctionCtx {
-                ty: FunctionType::EntryPoint(index as u16),
-                info: info.get_entry_point(index),
-                expressions: &ep.function.expressions,
-                named_expressions: &ep.function.named_expressions,
+        // Write all constants
+        for (handle, constant) in module.constants.iter() {
+            if constant.name.is_some() {
```

Comment:
The name here should be totally cosmetic. It shouldn't be used as a condition to emit the global constant.

Reply:
This mostly for readability. For example:
```wgsl
// vertex
let c_scale: f32 = 1.2;
return VertexOutput(uv, vec4<f32>(c_scale * pos, 0.0, 1.0));
``` 

If I remove this part of the code, `c_scale` will be inlined.

Reply:
I see. I suppose we can roll with this :)